pandas/tests/groupby/test_categorical.py

```diff
@@ -1252,3 +1252,29 @@ def test_get_nonexistent_category():
                 {"var": [rows.iloc[-1]["var"]], "val": [rows.iloc[-1]["vau"]]}
             )
         )
+
+
+@pytest.mark.parametrize("aggregation", ["sum", "mean", "min", "count"])
+@pytest.mark.parametrize("observed", [True, False])
+def test_series_groupby_on_2_categoricals_unobserved(aggregation: str, observed: bool):
+    # GH 17605
+    df = pd.DataFrame(
+        {
+            "cat_1": pd.Categorical(list("AABB"), categories=list("ABCD")),
+            "cat_2": pd.Categorical(list("AB") * 2, categories=list("ABCD")),
+            "value": [0.1] * 4,
+        }
+    )
+
+    # Expect 1 observation for each combination of categories
+    if observed:
+        expected_length = 4
+    else:
+        expected_length = len(df["cat_1"].cat.categories) * len(
+            df["cat_2"].cat.categories
+        )
+
+    series_groupby = df.groupby(["cat_1", "cat_2"], observed=observed)["value"]
+    agg = getattr(series_groupby, aggregation)
+    result = agg()
+    assert len(result) == expected_length
```

Comment:
I'd also like some kind of assertion about this values here. For example, I don't know at a glance whether `reindex_output` will fill with NaN or 0. For the case  of `.groupby(...).count()`, we'll want to ensure we have int dtype for the output.

You may need to have `expected_values` be part of the parameterization.

Reply:
Sure! I'm gonna parametrize this test with the `reduction_func` fixture as @WillAyd mentioned below, but I'll add some more specific test cases for `count` and the other aggregations that seem to be failing.